### DISCUSSION TITLE
Looking for opensource Identity middleware

### DISCUSSION THREAD

**Discussion by sanjaybharkatiya** [JanssenProject/jans#3870] [General]
We're seeking a highly scalable open-source middleware that can support a range of authentication and authorization capabilities, including passwordless authentication, multi-factor authentication, adaptive authentication, OIDC, OAuth, FIDO2, and SAML. The middleware should leverage our existing backend identity platform, which stores over a billion identities and devices in a NoSQL database, as well as our universal sign-in/sign-up page with account management functionality that's already live for both mobile and web applications.

We're interested in knowing if Janssen is a suitable solution to act as a middleware layer in this scenario.

**Comment by ossdhaval** [comment]
Hey @sanjaybharkatiya , it's good to see your post.

Yes, Janssen Server offers all the capabilities that you have listed above. Only for SAML support, we should probably understand your use-case in more detail. 

I am sure you would have visited Janssen Server documentation site already. I am sharing few links that describe what Janssen Server offers in much more details. Hope these links help.

- [Use-cases for Janssen Server](https://docs.jans.io/head/admin/planning/use-cases/)
- [Leveraging interception scripts](https://docs.jans.io/v1.0.6/admin/developer/interception-scripts/)
- [Identity Management](https://docs.jans.io/v1.0.6/admin/planning/identity-management/)

**Comment by nynymike** [comment]
Janssen can certainly support your use case. If you want a commercial support contract, you can [schedule a meeting with Gluu](https://gluu.org/booking)
